src/parser/jobs/sch/modules/FaerieGauge.js

```diff
@@ -0,0 +1,173 @@
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {ActionLink} from 'components/ui/DbLink'
+import TimeLineChart from 'components/ui/TimeLineChart'
+
+const SUMMON_ACTIONS = [
+	ACTIONS.SUMMON_EOS.id,
+	ACTIONS.SUMMON_SELENE.id,
+]
+// Actions that generate fairy gauge
+const GAUGE_GENERATORS = [
+	ACTIONS.SCH_ENERGY_DRAIN.id,
+	ACTIONS.LUSTRATE.id,
+	ACTIONS.INDOMITABILITY.id,
+	ACTIONS.SACRED_SOIL.id,
+	ACTIONS.EXCOGITATION.id,
+]
+// Gauge limits
+const GAUGE_MAX = 100
+const GAUGE_START = 0
+const GAUGE_GAIN_AMOUNT = 10
+// Graph color
+const GRAPH_COLOR = Color(JOBS.SCHOLAR.colour)
+const BG_COLOR_FADE = 0.8
+const BORDER_COLOR_FADE = 0.5
+
+export default class FaerieGauge extends Module {
+	static handle = 'fairieGauge'
+	static title = t('sch.gauge.title')`Faerie Gauge Usage`
+	static dependencies = [
+		'combatants',
+		'suggestions',
+	]
+
+	// Defaults
+	_gauge = GAUGE_START
+	_waste = 0
+	_history = []
+	_fairyOut = false
+	_noFairyAtStart = false
+
+	constructor(...args) {
+		super(...args)
+
+		// consumers
+		this.addHook('heal', {by: 'pet', abilityId: STATUSES.FEY_UNION.id}, this._onGaugeSpend)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.SCH_FEY_BLESSING.id}, this._onGaugeSpend)
+		this.addHook('death', {to: 'player'}, this._onDeath) // I mean.. it does consume all your gauge..
+		// generators
+		this.addHook('cast', {by: 'player', abilityId: GAUGE_GENERATORS}, this._onGaugeGenerate)
+		this.addHook('complete', this._onComplete)
+		// summoning a fairy
+		this.addHook('cast', {by: 'player', abilityId: SUMMON_ACTIONS}, this._onSummon)
+	}
+
+	// Search through the events to figure out if there was a fairy out before logs started
+	normalise(events) {
+		for (const event of events) {
+			if (!event.ability) { continue }
+
+			const action = getDataBy(ACTIONS, 'id', event.ability.guid)
+			if (!action) { continue }
+
+			// if the first action we find is a summon action, bail out and give a warning later
+			if (action.id && SUMMON_ACTIONS.includes(action.id)) {
+				this._noFairyAtStart = true
+				break
+			}
+
+			const pet = this.parser.report.friendlyPets.find(pet => pet.id === event.sourceID)
+				|| {petOwner: -1}
+
+			// Ignore events that aren't related to your fairy
+			if (
+				event.type !== 'cast' ||
+				!event.sourceIsFriendly ||
+				pet.petOwner !== this.parser.player.id ||
+				!action.pet
+			) { continue }
+
+			// Fairy found
+			this._fairyOut = true
+			break
+		}
+
+		return events
+	}
+
+	_onSummon() {
+		this._fairyOut = true
+	}
+
+	// both spenders and generators consume the same amount
+	_onGaugeSpend() {
+		this._gauge -= GAUGE_GAIN_AMOUNT
+		this._updateHistory()
+	}
+
+	_onGaugeGenerate() {
+		if (this.combatants.selected.hasStatus(STATUSES.DISSIPATION.id) || ! this._fairyOut) {
+			// can't generate a guage without the fairy, so bail out
+			return
```

Comment:
Should this track as waste for usage with no fairy out?  Especially when they're not under Dissipation, and just haven't resummoned the fairy after death, though I imagine there are cases where it's fine to burn off AF stacks while under Dissipation.

Reply:
Perhaps track in a separate variable? `this._noFairyWaste`, perhaps? I can see some situations where you come back from death and absolutely need the aetherflow and moves for immediate health and MP?

Reply:
Per our discussion on Discord, I’m ok with not tracking “waste” due to not having a fairy out, and just having a future recommendation if someone doesn’t resummon the fairy promptly after a death.

Reply:
Yeah, I agree. Seriously, though - thank you for the suggestions and input!